netlink-packet-route/src/rtnl/tc/nlas/action/mirred.rs

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+
+/// Mirred action
+///
+/// The mirred action allows packet mirroring (copying) or
+/// redirecting (stealing) the packet it receives. Mirroring is what
+/// is sometimes referred to as Switch Port Analyzer (SPAN) and is
+/// commonly used to analyze and/or debug flows.
+use crate::{
+    nlas::{self, DefaultNla, NlaBuffer},
+    tc::{constants::*, TC_GEN_BUF_LEN},
+    traits::{Emitable, Parseable},
+    DecodeError,
+};
+
+pub const KIND: &str = "mirred";
+pub const TC_MIRRED_BUF_LEN: usize = TC_GEN_BUF_LEN + 8;
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum Nla {
+    Unspec(Vec<u8>),
+    Tm(Vec<u8>),
+    Parms(TcMirred),
+    Other(DefaultNla),
+}
+
+impl nlas::Nla for Nla {
+    fn value_len(&self) -> usize {
+        use self::Nla::*;
+        match self {
+            Unspec(bytes) | Tm(bytes) => bytes.len(),
+            Parms(_) => TC_MIRRED_BUF_LEN,
+            Other(attr) => attr.value_len(),
+        }
+    }
+
+    fn emit_value(&self, buffer: &mut [u8]) {
+        use self::Nla::*;
+        match self {
+            Unspec(bytes) | Tm(bytes) => buffer.copy_from_slice(bytes.as_slice()),
+            Parms(p) => p.emit(buffer),
+            Other(attr) => attr.emit_value(buffer),
+        }
+    }
+    fn kind(&self) -> u16 {
+        use self::Nla::*;
+        match self {
+            Unspec(_) => TCA_MIRRED_UNSPEC,
+            Tm(_) => TCA_MIRRED_TM,
+            Parms(_) => TCA_MIRRED_PARMS,
+            Other(nla) => nla.kind(),
+        }
+    }
+}
+
+impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for Nla {
+    fn parse(buf: &NlaBuffer<&'a T>) -> Result<Self, DecodeError> {
+        use self::Nla::*;
+        let payload = buf.value();
+        Ok(match buf.kind() {
+            TCA_MIRRED_UNSPEC => Unspec(payload.to_vec()),
+            TCA_MIRRED_TM => Tm(payload.to_vec()),
+            TCA_MIRRED_PARMS => Parms(TcMirred::parse(&TcMirredBuffer::new_checked(payload)?)?),
+            _ => Other(DefaultNla::parse(buf)?),
+        })
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Default)]
+pub struct TcMirred {
+    // TcGen
```

Comment:
What does this comment mean? Is it just a leftover?

Reply:
The first 5 fields could use the public struct TcGen, but I haven't found a good implementation.

Removed first.